### PR TITLE
IBN-2419 make hue bridge read non blocking

### PR DESCRIPTION
**Pull request by Maw-Jung**
The call for the current hue bridge configuration was blocking and
didn't return prior to a timeout, which made the ui freeze. By
triggering a job to do just that, the ui keeps being responsive.

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java [hunk 1]
```diff
@@ -121,20 +128,23 @@ public void deactivate() {
...
+                onSensorAddedInternal(s);
+            }
+            List<Scene> scenes = hueBridgeHandler.getScenes();
+            for (Scene s : scenes) {
+                onSceneAddedInternal(s);
+            }
+            // search for unpaired lights
+            hueBridgeHandler.startSearch();
+        };
+        scheduler.schedule(command, DISCOVERY_SCAN_DELAY, SECONDS);
```
Comment (t-esch) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java]: Shouldn't this be `scheduler.execute(() => { ... } );`

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java [hunk 2]
```diff
@@ -74,6 +77,9 @@
     private final Logger logger = LoggerFactory.getLogger(HueLightDiscoveryService.class);
 
     private static final int SEARCH_TIME = 10;
+    private static final int DISCOVERY_SCAN_DELAY = 0;
+
+    private ScheduledExecutorService scheduler;
```
Comment (t-esch) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java]: The SonosDiscoveryService uses
`
private final ExecutorService pool = ThreadPoolManager.getPool(getClass().getSimpleName());
pool.submit(() -> { ... }
`

**Review by t-esch — CHANGES_REQUESTED** [review]
Should this included in 1.2.1650 or 1.2.1700?

**Comment by pwolfart** [comment]
I don`t think this is critical so lets place this in 1.2.1700.

**Pull request merged**